libflux/flux-core/src/semantic/tests.rs

```diff
@@ -3363,6 +3363,28 @@ fn function_default_arguments_and_pipes() {
     }
 }
 
+#[test]
+fn issue_4051() {
+    test_infer! {
+        env: map![
+            "r" => "A where A: Record",
+        ],
+        src: r#"
+        f = (r) => {
+            a = r.a
+            b = r.b + 1
+            c = r.c * 1.0
+            return r
+        }
+        x = f(r:r)
+        "#,
+        exp: map![
+            "f" => "(r:{S with a:X, b: int, c: float) => {S with a:X, b: int, c: float}",
```

Comment:
```suggestion
            "f" => "(r:{S with a:X, b: int, c: float}) => {S with a:X, b: int, c: float}",
```
I assume that there is a `}` missing here, but it still seems to parse?